### PR TITLE
Tutorials: updating the version to the Tribuo JAR files in the respective cells

### PR DESCRIPTION
**Pull request by neomatrix369**
### Description

Updating the version to the Tribuo JAR files in the Java-based tutorials to bring them in sync with the recently released 4.0.2 version

Also fixed JavaDoc links in the Regression Tutorial.

### Motivation

Raised as a fix on the back of #92

### CODE REVIEW

**Comment by Craigacp** [comment]
If you look at https://github.com/oracle/tribuo/commit/79b707eea0ddc5673cdd0e2d863b2c60b4f2d5d4 on the [v4.0.X-release-branch](https://github.com/oracle/tribuo/tree/v4.0.X-release-branch) you'll see the tutorials have the appropriate versions there.

**Comment by neomatrix369** [comment]
> If you look at [79b707e](https://github.com/oracle/tribuo/commit/79b707eea0ddc5673cdd0e2d863b2c60b4f2d5d4) on the [v4.0.X-release-branch](https://github.com/oracle/tribuo/tree/v4.0.X-release-branch) you'll see the tutorials have the appropriate versions there.

Oh I see, when is due to release? 

I also found the links to the JavaDocs were different (broken) - is there some auto-check to check in the tutorials and rest of the repo -- during some nightly CI/CD build process?

**Comment by neomatrix369** [comment]
Feel free to close the PR if this is redundant

**Comment by Craigacp** [comment]
As I said in the other thread, I plan to fix all the versions in the tutorials in `main` to use `4.1.0-SNAPSHOT` jars to reflect the version in main. If you want to do that with this PR that's ok, otherwise I'll do it a little later today.

**Comment by Craigacp** [comment]
> > If you look at [79b707e](https://github.com/oracle/tribuo/commit/79b707eea0ddc5673cdd0e2d863b2c60b4f2d5d4) on the [v4.0.X-release-branch](https://github.com/oracle/tribuo/tree/v4.0.X-release-branch) you'll see the tutorials have the appropriate versions there.
> 
> Oh I see, when is due to release?

4.1 doesn't have a fixed release date yet, we're leaving it a bit to see what things the community are interested in to make sure we catch any other features that people are after. There are already improvements in the tree packages, and an implementation of K-Means++. We'll also land something about rescaling regressions to make sure the models build correctly as this is a specific issue with LibSVM, and a multi-label linear SGD model. There are other fit and finish things around the RowProcessor that we're looking at, along with updating the backends to the latest/upcoming versions (e.g. ONNX Runtime 1.6, TF Java 0.2.0, XGBoost 1.3).

> 
> I also found the links to the JavaDocs were different (broken) - is there some auto-check to check in the tutorials and rest of the repo -- during some nightly CI/CD build process?

The javadoc link is an oversight on my part, we changed the javadoc URL during the initial deployment of the website vs the internal mirror we had and I forgot to update it in the tutorial.

**Comment by neomatrix369** [comment]
Kmeans++ sounds interesting.

I have applied the discussed changes across the notebooks. `4.1.0-SNAPSHOT` all across now.

File: tutorials/columnar-tribuo-v4.ipynb [hunk 1]
```diff
@@ -38,8 +38,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%jars ./tribuo-classification-experiments-4.1.0-SNAPSHOT-jar-with-dependencies.jar\n",
-    "%jars ./tribuo-json-4.1.0-SNAPSHOT-jar-with-dependencies.jar"
+    "%jars ./tribuo-classification-experiments-4.0.2-jar-with-dependencies.jar\n",
```
Comment (Craigacp) [inline tutorials/columnar-tribuo-v4.ipynb:41]: These should be left unchanged.

File: tutorials/configuration-tribuo-v4.ipynb [hunk 2]
```diff
@@ -32,8 +32,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%jars ./tribuo-classification-experiments-4.1.0-SNAPSHOT-jar-with-dependencies.jar\n",
-    "%jars ./tribuo-json-4.1.0-SNAPSHOT-jar-with-dependencies.jar"
+    "%jars ./tribuo-classification-experiments-4.0.2-jar-with-dependencies.jar\n",
```
Comment (Craigacp) [inline tutorials/configuration-tribuo-v4.ipynb:35]: These should be left unchanged.

File: tutorials/external-models-tribuo-v4.ipynb [hunk 3]
```diff
@@ -24,8 +24,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "%jars tribuo-classification-experiments-4.1.0-SNAPSHOT-jar-with-dependencies.jar\n",
-    "%jars tribuo-onnx-4.1.0-SNAPSHOT-jar-with-dependencies.jar"
+    "%jars tribuo-classification-experiments-4.0.2-jar-with-dependencies.jar\n",
```
Comment (Craigacp) [inline tutorials/external-models-tribuo-v4.ipynb:27]: These should be left unchanged.

**Review by Craigacp — CHANGES_REQUESTED** [review]
There are still a few versions set to 4.0.2 rather than 4.1.0-SNAPSHOT.

**Comment by Craigacp** [comment]
> Kmeans++ sounds interesting.

It's already integrated in `main` - https://github.com/oracle/tribuo/blob/183008bde8530ec540d4b644b278f903d4e11874/Clustering/KMeans/src/main/java/org/tribuo/clustering/kmeans/KMeansTrainer.java#L324

File: tutorials/configuration-tribuo-v4.ipynb [hunk 4]
```diff
@@ -817,7 +817,7 @@
       "\t\t\t\t\tnum-examples = 60000\n",
       "\t\t\t\t\tnum-features = 717\n",
       "\t\t\t\t\tnum-outputs = 10\n",
-      "\t\t\t\t\ttribuo-version = 4.1.0-SNAPSHOT\n",
+      "\t\t\t\t\ttribuo-version = 4.0.2\n",
```
Comment (Craigacp) [inline tutorials/configuration-tribuo-v4.ipynb:820]: Missed this one.

File: tutorials/configuration-tribuo-v4.ipynb [hunk 5]
```diff
@@ -845,7 +845,7 @@
       "\t\t\tinstance-values = Map{\n",
       "\t\t\t\treconfigured-model=true\n",
       "\t\t\t}\n",
-      "\t\t\ttribuo-version = 4.1.0-SNAPSHOT\n",
+      "\t\t\ttribuo-version = 4.0.2\n",
```
Comment (Craigacp) [inline tutorials/configuration-tribuo-v4.ipynb:848]: And this one.

**Review by Craigacp — CHANGES_REQUESTED** [review]

File: tutorials/configuration-tribuo-v4.ipynb (hunk 4 above)
Comment (neomatrix369) [reply tutorials/configuration-tribuo-v4.ipynb:820]: Fixed now

**Review by Craigacp — APPROVED** [review]
LGTM

**Comment by Craigacp** [comment]
Thanks!

**Pull request merged**